pkgs/games/shipwright/default.nix

```diff
@@ -0,0 +1,223 @@
+      pal-gc = "f788793d27aac3f8d91be5f242c4134217c615bfddd5c70384521ea2153435d2";
+      debug-mq = "161a3e8d6e67ecd5d996eb0251a946e6ae2b65089563ab1cd0873eca609b6a5f";
+    }.${romVariant} or (throw "Unsupported romVariant '${romVariant}'. Valid options are 'debug', 'pal-gc' or 'debug-mq'.");
+  }
+
+  # Compile with both a regular and MQ rom, useful for randomisers
+, enableDualMqRom ? false
+, ootMqRom ? requireFile {
+    name = "oot-debug-mq.z64";
+    sha256 = "161a3e8d6e67ecd5d996eb0251a946e6ae2b65089563ab1cd0873eca609b6a5f";
+    message = ''
+      This nix expression requires that oot-debug-mq.z64 is already part of the store.
+      To get this file you can dump your Ocarina of Time's cartridge to a file,
+      and add it to the nix store with nix-store --add-fixed sha256 <FILE>.
+
+      This error is thrown because you have enabled enableDualMqRom.
+    '';
+  }
+}:
+
+let
+  isMqVariant = lib.hasSuffix "mq" romVariant;
+in
+
+assert lib.assertMsg (!(isMqVariant && enableDualMqRom)) "The enableDualMqRom option is mutually exclusive with a master quest romVariant.";
+
+stdenv.mkDerivation rec {
+  pname = "shipwright";
+  version = "5.1.3";
+
+  src = fetchFromGitHub {
+    owner = "harbourmasters";
+    repo = "shipwright";
+    rev = version;
+    hash = "sha256-nDjaFrlBYXgxBW4BvsEI1uJVvXHgNwsro44B2Td85bI=";
+    fetchSubmodules = true;
+  };
+
+  enableParallelBuilding = true;
+
+  # Copied from the building guide
+  cmakeFlags = [
+    "DEBUG=0"
+    "OPTFLAGS=-O2"
+  ];
+
+  # Linking fails without this
+  hardeningDisable = [ "all" ];
+
+  # Many warnings are being treated as errors
+  NIX_CFLAGS_COMPILE = [
+    "-Wno-error"
+  ];
+
+  nativeBuildInputs = [
+    python3
+    llvmPackages.bintools
+    copyDesktopItems
+    cmake
+    ninja
+    pkg-config
+    lsb-release
+    clang
+    imagemagick
+    makeWrapper
+  ];
+
+  buildInputs = [
+    libX11
+    libXrandr
+    libXinerama
+    libXcursor
+    libXi
+    libXext
+    glew
+    SDL2
+    SDL2_net
+    libpulseaudio
+    libpng
+    bzip2
+    StormLib
+  ];
+
+  postPatch = ''
+    # This script attempts to call git to retrieve the commit hash,
+    # so we have to manually patch it in instead.
+    substituteInPlace ZAPDTR/ZAPD/genbuildinfo.py \
+      --replace "label = subprocess" "#label = subprocess" \
+      --replace "+ label +" "+ \"${builtins.substring 0 7 src.rev}\" +"
+
+    # SHIPWRIGHT: Stop ship from looking at user dir for OTRs
+    substituteInPlace soh/soh/OTRGlobals.cpp \
+      --replace "std::string mqPath = Ship::Window::GetPathRelativeToAppDirectory(\"oot-mq.otr\");" "std::string mqPath = \"oot-mq.otr\";" \
+      --replace "std::string ootPath = Ship::Window::GetPathRelativeToAppDirectory(\"oot.otr\");" "std::string ootPath = \"oot.otr\";"
+
+    # Copy Rom(s) to OTRExporter for asset generation.
+    ln -s ${ootRom} OTRExporter/oot.z64
+    ${lib.optionalString enableDualMqRom "ln -s ${ootMqRom} OTRExporter/oot-mq.z64" }
```

Comment:
If `ootRom`is a master quest variant, shouldn't it be copied to `oot-mq.64` as well?